main.go

```diff
@@ -19,28 +20,80 @@ func main() {
 	ctx := context.Background()
 	ctx, cancel := context.WithCancel(ctx)
 
-	// Trap SIGINT to trigger eqReceiptProcessor graceful shutdown.
+	// Trap SIGINT to trigger graceful shutdown
 	signals := make(chan os.Signal, 1)
 	signal.Notify(signals, os.Interrupt)
 
-	processors := StartProcessors(ctx, appConfig)
+	// Channel for goroutines to notify main of errors
+	errChan := make(chan error)
 
-	// block until we receive eqReceiptProcessor shutdown signal
+	processors, err := StartProcessors(ctx, appConfig, errChan)
+	if err != nil {
+		shutdown(ctx, cancel, processors)
+	}
+	// Indicate readiness
+	err = readiness.Ready(ctx, appConfig.ReadinessFilePath)
```

Comment:
the naming of this makes it look like this is only being used for readiness checking, but it's also being used for the liveness check also. maybe we should be "touching" the file every minute (or maybe less) to indicate some kind of minimal liveness